docs/StardustDocs/topics/dataSources/Parquet.md

```diff
@@ -0,0 +1,192 @@
+public fun DataFrame.Companion.readParquet(
+    vararg strUrls: String,
+    nullability: NullabilityOptions = NullabilityOptions.Infer,
+    batchSize: Long = ARROW_PARQUET_DEFAULT_BATCH_SIZE,
+): AnyFrame
+
+// 3) Paths
+public fun DataFrame.Companion.readParquet(
+    vararg paths: Path,
+    nullability: NullabilityOptions = NullabilityOptions.Infer,
+    batchSize: Long = ARROW_PARQUET_DEFAULT_BATCH_SIZE,
+): AnyFrame
+
+// 4) Files
+public fun DataFrame.Companion.readParquet(
+    vararg files: File,
+    nullability: NullabilityOptions = NullabilityOptions.Infer,
+    batchSize: Long = ARROW_PARQUET_DEFAULT_BATCH_SIZE,
+): AnyFrame
+```
+
+These overloads are defined in the `dataframe-arrow` module and internally use `FileFormat.PARQUET` from Apache Arrow’s
+Dataset API to scan the data and materialize it as a Kotlin `DataFrame`.
+
+### Examples
+
+```kotlin
+// Read from file paths (as strings)
+val df1 = DataFrame.readParquet("data/sales.parquet")
+
+// Read from File objects
+val file = File("data/sales.parquet")
+val df2 = DataFrame.readParquet(file)
+
+// Read from Path objects
+val path = Paths.get("data/sales.parquet")
+val df3 = DataFrame.readParquet(path)
+
+// Read from URLs
+val url = URL("https://example.com/data/sales.parquet")
+val df4 = DataFrame.readParquet(url)
+
+// Customize nullability inference and batch size
+val df5 = DataFrame.readParquet(
+    File("data/sales.parquet"),
+    nullability = NullabilityOptions.Infer,
+    batchSize = 64L * 1024 // tune Arrow scan batch size if needed
+)
+```
+
+### Multiple Files
+
+It's possible to read multiple Parquet files:
+
+```kotlin
+val df = DataFrame.readParquet("file1.parquet", "file2.parquet", "file3.parquet")
+```
+**Requirements:**
+
+- All files must have compatible schemas
+- Files are vertically concatenated (union of rows)
+- Column types must match exactly
+- Missing columns in some files will result in null values
+
+### Batch Size Tuning
+
+- **Default**: (typically 1024) `ARROW_PARQUET_DEFAULT_BATCH_SIZE`
+- **Small files** (< 100MB): Use default
+- **Large files** (> 1GB): Increase to `64 * 1024` or `128 * 1024`
+- **Memory constrained**: Decrease to `256` or `512`
+
+```kotlin
+// For large files with enough memory
+DataFrame.readParquet("large_file.parquet", batchSize = 64L * 1024)
+
+// For memory-constrained environments  
+DataFrame.readParquet("file.parquet", batchSize = 256L)
+```
+
+### Nullability Inference
+
+Controls how nullable columns are handled:
+
+```kotlin
+// Infer nullability from data (default)
+DataFrame.readParquet("file.parquet", nullability = NullabilityOptions.Infer)
+
+// Treat all columns as nullable
+DataFrame.readParquet("file.parquet", nullability = NullabilityOptions.Enable)
+
+// Treat all columns as non-null (may cause runtime errors)
+DataFrame.readParquet("file.parquet", nullability = NullabilityOptions.Disable)
+```
+
+## About Parquet
```

Comment:
doesn't add much to the DataFrame documentation. We don't have to sell Parquet to people, right? People are likely already using Parquet if they are reading this page